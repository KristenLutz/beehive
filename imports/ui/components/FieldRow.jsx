// @flow

// Framework
import React from "react";

// Components
import { Container, Row, Col, FormGroup, Label, Input } from "reactstrap";

// Helpers
import { camelCaseToTitle } from "../helpers/text";

// Types
type InputValue = string | boolean | number;

/**
 * A single input field for a boolean, string or number.
 *
 * @param {Object} props All the properties passed to the React Component.
 * @returns {Element} Stateless functional React component.
 */
const FieldRow = ({
  label = "Field",
  value,
  visibleLabel = true,
  stateSettingAction
}: {
  label: string,
  value: InputValue,
  stateSettingAction: ((label: string, value: InputValue) => void) | void,
  visibleLabel: boolean | void
}): StatelessComponent => {
  const inputId: string = `${label}-${typeof value}-field-row-input`;
  const titleLabel: string = camelCaseToTitle(label);
  const labelClass: string = visibleLabel ? "" : "sr-only";
  const onChange = ({ target }) => stateSettingAction(label, target.value);
  return (
    <FormGroup className="field-row">
      <Label className={labelClass} for={inputId}>
        {titleLabel}
      </Label>
      <Input
        type="text"
        name={inputId}
        id={inputId}
        onChange={onChange}
        value={value}
      />
    </FormGroup>
  );
};

export default FieldRow;
