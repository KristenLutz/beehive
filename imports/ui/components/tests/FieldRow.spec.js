import React from "react";
import ReactTestRenderer from "react-test-renderer";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-15";

import FieldRow from "../FieldRow";
import { Input, Label } from "reactstrap";

configure({ adapter: new Adapter() });

describe("<FieldRow />", () => {
  const defaultProps = {
    label: "NewField",
    value: 15,
    visibleLabel: false,
    stateSettingAction: jest.fn()
  };

  describe("FieldRow SnapShot", () => {
    it("renders FieldRow correctly", () => {
      const tree = ReactTestRenderer.create(
        <FieldRow {...defaultProps} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("FieldRow Props", () => {
    let wrapper, input, label;

    beforeEach(() => {
      const wrapper = shallow(<FieldRow {...defaultProps} />);
      input = wrapper.find(Input);
    });

    it("should pass fields prop to FieldRows", () => {
      input.simulate("change", { target: { value: "My new value" } });
      expect(defaultProps.stateSettingAction).toHaveBeenCalled();
    });
  });
});
