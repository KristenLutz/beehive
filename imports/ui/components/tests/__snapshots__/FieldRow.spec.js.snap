// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldRow /> FieldRow SnapShot renders FieldRow correctly 1`] = `
<div
  className="field-row form-group"
>
  <label
    className="sr-only  form-control-label"
    htmlFor="NewField-number-field-row-input"
  >
    New Field
  </label>
  <input
    className="form-control"
    id="NewField-number-field-row-input"
    name="NewField-number-field-row-input"
    onChange={[Function]}
    placeholder="New Field"
    type="text"
    value={15}
  />
</div>
`;
